app/helpers/application_helper/button/middleware_instance_add.rb

```diff
@@ -0,0 +1,13 @@
+class ApplicationHelper::Button::MiddlewareInstanceAdd < ApplicationHelper::Button::MiddlewareAction
```

Comment:
The name "MiddlewareInstanceAdd" is little bit cryptic to me, not only because it's used also in case of actions that actually removes stuff from the server. The only difference between this and `MiddlewareAction` is that this button is hidden if the related `@record` is a server in the domain mode or an entity under such a server. So what about something like "HiddenForDomain" or "StandaloneServerAction" vs "DomainServerAction".

Hmm, probably `StandaloneServerAction < ServerAction`, `DomainServerAction < ServerAction` class hierarchy would be the best, because I have a use case in which I need to enable certain action only for domain server (it needs to be disabled for the standalone (/normal) server)

`ServerAction` ... basically your MiddlewareAction, i.e. all the servers or server entities w/o the product set to 'Hawkular'
`StandaloneServerAction` ... your `MiddlewareInstanceAdd`
`DomainServerAction` ... the opposite of `MiddlewareInstanceAdd` (I can create that class in a followup PR, because there is no use case for it now, but it'll be)